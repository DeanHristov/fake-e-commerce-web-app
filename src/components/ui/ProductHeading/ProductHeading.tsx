import { FC, ReactNode } from 'react';
import { useRouter } from 'next/navigation';

export interface IProductHeadingProps {
  id: number;
  title: string;
  children?: ReactNode;
  className?: string;
}

const ProductHeading: FC<IProductHeadingProps> = ({
  id,
  title,
  children,
  className,
}) => {
  const router = useRouter();

  return (
    <h3
      onClick={() => router.push(`/products/${id}`)}
      className={`text-black text-lg mt-2 truncate hover:underline hover:cursor-pointer ${className}`}
    >
      {children}
      {title}
    </h3>
  );
};

export default ProductHeading;
