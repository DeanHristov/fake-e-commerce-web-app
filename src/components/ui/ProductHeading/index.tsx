import ProductHeading, { IProductHeadingProps } from './ProductHeading';

export type { IProductHeadingProps };
export default ProductHeading;
