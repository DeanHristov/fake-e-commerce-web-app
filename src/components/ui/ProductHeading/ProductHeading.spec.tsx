import { render } from '@testing-library/react';

import ProductHeading from './';

describe('UI/ProductHeading <ProductHeading {...} />', () => {
  it('Should render a default state of the Component', () => {
    const { container } = render(<ProductHeading title={'Product 1'} id={1} />);

    expect(container).toMatchSnapshot();
  });
});
