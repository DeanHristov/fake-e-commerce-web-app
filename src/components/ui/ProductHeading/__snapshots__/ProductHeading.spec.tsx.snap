// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/ProductHeading <ProductHeading {...} /> Should render a default state of the Component 1`] = `
<div>
  <h3
    class="text-black text-lg mt-2 truncate hover:underline hover:cursor-pointer undefined"
  >
    Product 1
  </h3>
</div>
`;
